website/source/docs/platform/k8s/helm.html.md

```diff
@@ -377,6 +377,176 @@ to run the sync program.
+      This is useful if node IPs are routable from other DCs.
+      `useNodeName` and `host` must be `false` and `""` respectively.
+
+      * <a name="v-meshgateway-wanaddress-usenodename" href="#v-meshgateway-wanaddress-usenodename">`useNodeName`</a> (`bool: false`) -
+      If true, each Gateway Pod will advertise its Node Name
+      (as provided by the Kubernetes downward API) as the wan address.
+      This is useful if the node names are DNS entries that are routable from other DCs.
+      `useNodeIP` and `host` must be `false` and `""` respectively.
+
+      * <a name="v-meshgateway-wanaddress-host" href="#v-meshgateway-wanaddress-host">`host`</a> (`string: """`) -
+      If non-empty, each gateway Pod will use this host as its wan address.
+      Users must ensure that this address routes to the Gateway pods,
+      for example via a load balancer IP or a DNS entry that routes to the Service
+      fronting the Deployment.
+      `meshGateway.wanAddress.port` will be used as the port for the wan address.
+      `useNodeIP` and `useNodeName` must be false.
+
+  * <a name="v-meshgateway-service" href="#v-meshgateway-service">`service`</a> -
+  Values that configure the Kubernetes `Service` for the gateway `Deployment.
+
+      * <a name="v-meshgateway-service-enabled" href="#v-meshgateway-service-enabled">`enabled`</a> (`bool: false`) -
+      Whether to create a Service or not.
+
+      * <a name="v-meshgateway-service-type" href="#v-meshgateway-service-type">`type`</a> (`string: "ClusterIP"`) -
+      Service `type`, ex. `NodePort`, `LoadBalancer`, `ClusterIP`.
+
+      * <a name="v-meshgateway-service-port" href="#v-meshgateway-service-port">`port`</a> (`int: 443`) -
+      Port that the service will expose.
+      `targetPort` will automatically be set to `meshGateway.containerPort`.
+
+      * <a name="v-meshgateway-service-nodeport" href="#v-meshgateway-service-nodeport">`nodePort`</a> (`int`) -
+      Optional [`nodePort`](https://kubernetes.io/docs/concepts/services-networking/service/#nodeport) of the `Service`. Can be used in conjunction with
+      `type: NodePort`.
+
+      * <a name="v-meshgateway-service-annotations" href="#v-meshgateway-service-annotations">`annotations`</a> (`string`) -
+      Optional YAML string for additional annotations on the Service. This should be a formatted as a multi-line string.
+
+        ```yaml
+        annotations: |
+          "sample/annotation1": "foo"
+          "sample/annotation2": "bar"
+        ```
+
+      * <a name="v-meshgateway-service-additionalspec" href="#v-meshgateway-service-additionalspec">`additionalSpec`</a> (`string`) -
+      Optional string for appending additional YAML to the Service spec. This should be a formatted as a multi-line string.
+
+        ```yaml
+        additionalSpec: |
+          clusterIP: 10.0.171.239
+        ```
+
+  * <a name="v-meshgateway-imageenvoy" href="#v-meshgateway-imageenvoy">`imageEnvoy`</a> (`string: "envoyproxy/envoy:v1.10.0"`) -
```

Comment:
For my own info: We aren't providing a way for users to pick their Gateway proxy (like we do for sidecars)? Is this on the roadmap?

Reply:
Only envoy is supported right now. Not sure if other proxies are on the roadmap.